release-notes/swan-lake-2201.1.0-release-note.md

```diff
@@ -30,9 +30,386 @@ If you have not installed Ballerina, then download the [installers](/downloads/#
+    io:println(decimals:round(x));	// 6
+    io:println(decimals:round(5.55, fractionDigits = 0));	// 6
+    io:println(decimals:round(5.5565, fractionDigits = 3));	// 5.556
+}
+```
+
+##### An unreachable panic statement now does not result in a compilation error.
+
+```ballerina
+function fn() returns string {
+    int|string a = 10;
+
+    if a is int {
+        return "INT";
+    } else {
+        return "STRING";
+    }
+
+    panic error("Not Reached!"); // unreachable, but not an error.
+}
+```
+
+#### Bug Fixes
+
+##### Invalid subtype relation in table and anydata
+Fixed the subtype relation between table and anydata
+
+```ballerina
+type TANY table<map<any>>;
+
+public function main() {
+    TANY tany = table [{"a": 2}];
+    anydata _ = tany; // This is a compile error
+}
+```
+
+
+##### Fix Invalid value space when a union of singletons contains the null literal
+The value of null literal should be set as `null`.
+
+```ballerina
+type Foo boolean|null;
+
+public function main() {
+    Foo _ = "null"; // compilation error
+    boolean|null _ = "null"; // compilation error
+}
+```
+
+##### Fix issue in enum member value when member name is a quoted identifier
+
+Fixed the quote is included in enum member value when member name is a quoted identifier.
+
+```ballerina
+import ballerina/io;
+
+public enum MyEnum {
+    'new
+}
+
+public function main() {
+    io:println('new); // Previously prints `'new` now prints `new`
+}
+```
+
+
+##### Fix few issues related match pattern parsing
+
+- No syntax error when there is an extra comma inside mapping match pattern has been fixed.
+
+```ballerina
+type MyRecord record {
+    int field1;
+    int field2;
+};
+
+function fn(MyRecord r1) {
+    match r1 {
+        {field1: 0} => { // A syntax error is now given for the comma.
+
+        }
+    }
+}
+```
+
+- An issue parsing qualified identifier with predeclared prefix as const match pattern has been fixed.
+
+```ballerina
+function fn(any x) {
+    match x {
+        int:MAX_VALUE => { // Match pattern is now allowed
+        }
+
+        [int:MAX_VALUE, int:MIN_VALUE] => { // Match pattern is now allowed
+        }
+    }
+}
+```
+
+- Qualified identifiers being not allowed in error match pattern has been fixed.
+
+```ballerina
+function fn(error e) {
+    match e {
+        error(myError:MSG_1) => { // Match pattern is now allowed
+
+        }
+    }
+}
+```
+
+##### Fix compiler crashing when a query expression that throws an error is enclosed within a statement with on-fail. 
+
+```ballerina
+function verifyCheck() returns error? {
+    return error("custom error");
+}
+
+public function main() {
+    do {
+        _ = from int v in 1 ... 3
+            select check verifyCheck();
+    } on fail error err {
+        io:println("error caught", err);
+    }
+}
+```
+
+##### The inherent type for any arr = [1]; should be (any|error)[]
+
+For any arr = []; Earlier, the inherent type is any[] but it is now corrected to be (any|error)[]
```

Comment:
```suggestion
- The inherent type of a list constructed using a list constructor with `any` as the contextually-expected type has been fixed to be `(any|error)[]` instead of `any[]`.
```